src/braket/circuits/circuit.py

```diff
@@ -773,6 +805,98 @@ def apply_initialization_noise(
 
         return apply_noise_to_moments(self, noise, target_qubits, "initialization")
 
+    def make_bound_circuit(self, param_values: Dict[str, Number], strict: bool = False) -> Circuit:
+        """
+        Binds FreeParameters based upon their name and values passed in. If parameters
+        share the same name, all the parameters of that name will be set to the mapped value.
+
+        Args:
+            param_values (Dict[str, Number]):  A mapping of FreeParameter names
+                to a value to assign to them.
+            strict (bool, optional): If True, raises a ValueError if none of the FreeParameters
+                in param_values appear in the circuit. False by default."
+
+        Returns:
+            Circuit: Returns a circuit with all present parameters fixed to their respective
+            values.
+        """
+        if strict:
+            self._validate_parameters(param_values)
+        return self._use_parameter_value(param_values)
+
+    def _validate_parameters(self, parameter_values: Dict[str, Number]):
+        """
+        This runs a check to see that the parameters are in the Circuit.
+
+        Args:
+            parameter_values (Dict[str, Number]):  A mapping of FreeParameter names
+                to a value to assign to them.
+
+        Raises:
+            ValueError: If there are no parameters that match the key for the arg
+            param_values.
+        """
+        parameter_strings = set()
+        for parameter in self.parameters:
+            parameter_strings.add(str(parameter))
+        for param in parameter_values:
+            if param not in parameter_strings:
+                raise ValueError(f"No parameter in the circuit named: {param}")
+
+    def _use_parameter_value(self, param_values: Dict[str, Number]) -> Circuit:
+        """
+        Creates a Circuit that uses the parameter values passed in.
+
+        Args:
+            param_values (Dict[str, Number]): A mapping of FreeParameter names
+                to a value to assign to them.
+
+        Returns:
+            Circuit: A Circuit with specified parameters swapped for their
+            values.
+
+        """
+        fixed_circ = Circuit()
+        for instruction in self.instructions:
+            if self._check_for_params(instruction):
+                params = instruction.operator.parameters
+                bound_mapping = dict(
+                    map(
+                        lambda x: (str(x), param_values[str(x)])
+                        if str(x) in param_values
+                        else (str(x), x),
+                        params,
+                    )
+                )
```

Comment:
Why does this need to happen per operation? From what I can tell, this is what's happening:
1. `param_values` is filtered to include parameters that apply to the current operation
2. The parameters _not_ being assigned values from `param_values` are thrown in as well and mapped to their current values.
3. The resulting `bound_mapping` dict, containing every parameter on the `operation`, has its values validated and is then bound to the operation.

Why can't all the values in `parameter_values` be validated just once, and then for each operation, the entire `parameter_values` dict is bound? After all, only the relevant parameters are selected.